### DISCUSSION TITLE
Error when trying to download a backup

### DISCUSSION THREAD

**Discussion by carlisia** [vmware-tanzu/velero#3420] [Community support Q&A] [answered]
Running the command `velero backup download <backup name>` and getting:

```
An error occurred: Get "http://minio.velero.svc:9000/velero/backups/test1/test1.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20210208%2Fminio%2Fs3%2Faws4_request&X-Amz-Date=20210208T145945Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=1b5aafa73c614741cb20f3cc098e5257e1c19e6111599c53ff8f9fb21ffba7a4": dial tcp 10.195.47.241:9000: connect: connection refused
```

**Comment by zubron** [comment, marked as answer]
This error is caused by the MinIO service running as an instance inside your cluster (don't do this for production!) but not being accessible outside the cluster. 

To fix this, you will need to forward the MinIO port to the local machine using `kubectl port-forward -n <velero-namespace> svc/minio 9000` and then update the BSL configuration to use that as its "public URL" by adding `publicUrl: http://localhost:9000` to the BSL config. 

More details about the `publicUrl` config field can be found in the AWS plugin docs: https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/main/backupstoragelocation.md
